lib/model/channel.dart

```diff
@@ -139,6 +140,25 @@ mixin ChannelStore on UserStore {
     }
   }
 
+  bool selfHasContentAccess(ZulipStream channel) {
+    // Compare web's stream_data.has_content_access.
+    if (channel.isWebPublic) return true;
+    if (channel is Subscription) return true;
+    // Here web calls has_metadata_access... but that always returns true,
+    // as its comment says.
+    if (selfUser.role == UserRole.guest) return false;
+    if (!channel.inviteOnly) return true;
+    return _selfHasContentAccessViaGroupPermissions(channel);
+  }
+
+  bool _selfHasContentAccessViaGroupPermissions(ZulipStream channel) {
```

Comment:
```
channel: Add selfHasContentAccess, using canSubscribeGroup et al

[…]

This particular permission is one we don't yet have any logic that
should consume […]
```

I think it does actually :) see

- #1786

which has corresponding `TODO(#1786)`s in the code.